apps/global_full/4C_global_full_main.cpp

```diff
@@ -343,14 +344,35 @@ int main(int argc, char* argv[])
       input_file.emit_metadata(std::cout);
     }
   }
-  else if ((argc == 4) && (strcmp(argv[1], "--to-yaml") == 0))
+  else if ((argc >= 3) && (strcmp(argv[1], "--to-yaml") == 0))
   {
     if (Core::Communication::my_mpi_rank(lcomm) == 0)
     {
+      std::filesystem::path inputfile_name = argv[2];
+      std::filesystem::path outputfile_name;
       Core::IO::InputFile input_file = Global::set_up_input_file(lcomm);
-      input_file.read(argv[2]);
-      std::ofstream output_file(argv[3]);
+      input_file.read(inputfile_name);
+      if (argc == 3)
+      {
+        outputfile_name = inputfile_name.replace_extension("yaml");
```

Comment:
```suggestion
        outputfile_name = inputfile_name.replace_extension("4C.yaml");
```
Should the extension be `4C.yaml`?

Reply:
why? If I have <problem_name>.yaml, what else could that be?

Reply:
@ischeider We should use `.4C.yaml` to indicate that this file follows the 4C schema. See also the TGM where Laura shows this in the live demo.

Reply:
@ischeider I think you still need to change that to `.4C.yaml` here?